hate-speech-api/hate_speech_clf_api.py

```diff
@@ -0,0 +1,80 @@
+import sys
+import pickle
+from flask import Flask, jsonify, abort, request
+from flask_classful import FlaskView, route
+import data_processing.mm_segmenter as mm
+from data_processing.mm_converter import zawgyi_to_unicode
+import re
+import json
+from sklearn.ensemble import RandomForestClassifier
+
+app = Flask(__name__)
+
+class HateSpeechClassifier(FlaskView):
+
+    def __init__(self):
+        self.model = "None"
+        self.burmese_segmenter = mm.Segmenter()
+        self.clf_mapping = {
+            "RF" : "clfs/oversampled-RFClf.pkl"
+        }
+        with open("clfs/oversampled-tfidf-vect.pkl", 'rb') as vect_b:
+            self.vectorizer = pickle.load(vect_b)
+        with open("clfs/oversampled-RFClf.pkl", 'rb') as clf_b:
+            self.model = pickle.load(clf_b)
+
+    #Internal functions. Not exposed via API    
+    def _preprocess(self, text_document):
+        non_burmese_pattern = re.compile("[^"u"\U00001000-\U0000109F"u"\U00000020""]+", flags=re.UNICODE)
+        try:
+            preprocessed_text_document = re.sub("\s+", ' ', non_burmese_pattern.sub('', text_document)) 
+            if len(preprocessed_text_document) ==0 :
+                return ""
+            preprocessed_text_document = zawgyi_to_unicode(preprocessed_text_document)
+            preprocessed_text_document = " ".join(self.burmese_segmenter.segment(preprocessed_text_document))
+        except ValueError as e:
+            abort(400)
+        return preprocessed_text_document
+
+    #API endpoints
+    @route("/preprocess", methods=["POST"])
```

Comment:
what's the difference between this and _preprocess()?

Reply:
Since predict api endpoint also needs to preprocess the data, I decided to make an internal _preprocess function that can be used by both preprocess endpoint and predict endpoint. Calling a function (preprocess) from an endpoint was easier to implement than calling preprocess endpoint from predict endpoint...also more efficient I think.